storage/src/main/scala/coop/rchain/storage/package.scala

```diff
@@ -65,18 +67,25 @@ package object storage {
         s"consume: searching for data matching <patterns: $patterns> at <channels: $channels>")
       extractDataCandidates(store, channels, patterns)(txn) match {
         case None =>
-          store.putK(txn, channels, patterns, continuation)
+          store.putK(txn, channels, WaitingContinuation(patterns, continuation, persist))
           for (channel <- channels) store.addJoin(txn, channel, channels)
           logger.info(
             s"consume: no data found, storing <(patterns, continuation): ($patterns, $continuation)> at <channels: $channels>")
           None
         case Some(dataCandidates) =>
           dataCandidates.foreach {
-            case DataCandidate(candidateChannel, _, dataIndex) =>
+            case DataCandidate(candidateChannel, Datum(_, persistData), dataIndex)
+                if !persistData =>
               store.removeA(txn, candidateChannel, dataIndex)
+            case _ =>
+              ()
+          }
+          if (persist) {
```

Comment:
This won't work. If a persistent consume gets a match, the client code should get the continuation, and then put it back until it doesn't get a match any more.

Otherwise if a persistent read matches multiple inputs, it will only get one.